mitiq/lre/tests/test_layerwise_folding.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Unit tests for scaling noise by unitary folding of layers in the input
+circuit to allow for multivariate extrapolation."""
+
+from copy import deepcopy
+
+import pytest
+from cirq import Circuit, LineQubit, ops
+
+from mitiq.lre.multivariate_scaling.layerwise_folding import (
+    _get_chunks,
+    _get_num_layers_without_measurements,
+    _get_scale_factor_vectors,
+    multivariate_layer_scaling,
+)
+
+qreg1 = LineQubit.range(3)
+test_circuit1 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+    [ops.X.on(qreg1[2])],
+    [ops.TOFFOLI.on(*qreg1)],
+)
+
+test_circuit1_with_measurements = deepcopy(test_circuit1)
+test_circuit1_with_measurements.append(ops.measure_each(*qreg1))
+
+
+def test_multivariate_layerwise_scaling():
+    """Checks if multiple scaled circuits are returned to fit the required
+    folding pattern for multivariate extrapolation."""
+    multiple_scaled_circuits = multivariate_layer_scaling(
+        test_circuit1, 2, 2, 3
+    )
+
+    assert len(multiple_scaled_circuits) == 10
+    folding_pattern = [
+        (1, 1, 1),
+        (5, 1, 1),
+        (1, 5, 1),
+        (1, 1, 5),
+        (9, 1, 1),
+        (5, 5, 1),
+        (5, 1, 5),
+        (1, 9, 1),
+        (1, 5, 5),
+        (1, 1, 9),
+    ]
+
+    for i, scale_factor_vector in enumerate(folding_pattern):
+        scale_layer1, scale_layer2, scale_layer3 = scale_factor_vector
+        expected_circuit = Circuit(
+            [ops.H.on_each(*qreg1)] * scale_layer1,
+            [ops.CNOT.on(qreg1[0], qreg1[1]), ops.X.on(qreg1[2])]
+            * scale_layer2,
+            [ops.TOFFOLI.on(*qreg1)] * scale_layer3,
+        )
+        assert expected_circuit == multiple_scaled_circuits[i]
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [(test_circuit1, 3), (test_circuit1_with_measurements, 3)],
+)
+def test_get_num_layers(test_input, expected):
+    """Verifies function works as expected."""
+    calculated_num_layers = _get_num_layers_without_measurements(test_input)
+
+    assert calculated_num_layers == expected
+
+
+@pytest.mark.parametrize(
+    "test_input, test_chunks, expected_chunks",
+    [
+        (test_circuit1 + test_circuit1 + test_circuit1, 3, 3),
+        (test_circuit1 + test_circuit1 + test_circuit1, 1, 1),
+        (test_circuit1 + test_circuit1 + test_circuit1, 5, 5),
+    ],
+)
+def test_get_num_chunks(test_input, test_chunks, expected_chunks):
+    """Verifies the chunking function works as expected."""
+    calculated_num_chunks = len(_get_chunks(test_input, test_chunks))
```

Comment:
Isn't the expectation here as simple as 
```
assert test_chunks = len(_get_chunks(test_input, test_chunks))
```
If so, we can get rid of the `expected_chunks` parameter and the test becomes easier to read it logically.